lib/iex/lib/iex/evaluator.ex

```diff
@@ -78,25 +80,42 @@ defmodule IEx.Evaluator do
     )
   end
 
-  def parse(input, opts, buffer) do
+  def parse(input, opts, {buffer, last_op}) do
     input = buffer <> input
     file = Keyword.get(opts, :file, "nofile")
     line = Keyword.get(opts, :line, 1)
     column = Keyword.get(opts, :column, 1)
     charlist = String.to_charlist(input)
 
     result =
-      with {:ok, tokens} <- :elixir.string_to_tokens(charlist, line, column, file, opts) do
-        :elixir.tokens_to_quoted(adjust_operator(tokens, line, column, file, opts), file, opts)
+      with {:ok, tokens} <- :elixir.string_to_tokens(charlist, line, column, file, opts),
+           {:ok, adjusted_tokens} <- adjust_operator(tokens, line, column, file, opts, last_op),
+           {:ok, forms} <- :elixir.tokens_to_quoted(adjusted_tokens, file, opts) do
+        last_op =
+          case forms do
+            {:=, _, [_, _]} -> :match
+            _ -> :other
+          end
+
+        {:ok, forms, last_op}
       end
 
     case result do
-      {:ok, forms} ->
-        {:ok, forms, ""}
+      {:ok, forms, last_op} ->
+        {:ok, forms, "", last_op}
```

Comment:
Now we should change this to be `{:ok, forms, {"", last_op}}`. The API on server should not change but we should probably rename the `buffer` variable in there to something else. Maybe `parser_state`?

Reply:
> Maybe `parser_state`?

Luckily enough, we pass the necessity to name things here; it makes sense to restructure it everywhere, so `buffer` stays as in `{buffer, last_op}`.